### PR TITLE
API HW 2 and 3

### PR DESCRIPTION
**Pull request by katherinaGo**

### CODE REVIEW

File: EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "BiblesBaseUrl": "https://api.scripture.api.bible",
+  "PhonesBaseUrl": "https://api.restful-api.dev",
+  "ApiKey": ""
```
Comment (dnnyhlwdplc) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/appsettings.json:4]: oh, you are attentive to details! well done :)

File: EpamWebSiteTestingFramework/Epam.TestAutomation.API/BaseController.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    protected (RestResponse Response, T ResponseModel) Update<T, TPayload>(string resource, TPayload? payload = null)
+        where TPayload : class
+    {
+        var request = new RestRequest(resource, Method.Put);
+        if (payload != null)
+            request.AddJsonBody(payload);
+
+        var response = _restClient.ExecutePost(request);
```
Comment (dnnyhlwdplc) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.API/BaseController.cs]: ExecutePost?

File: EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/PhoneCreationTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            {
+                price = 700,
+                year = 2022
+            }
+        };
+
+        var createdPhone = new PhoneController(new CustomRestClient()).AddPhone<Phone>(phoneToCreate).Phone;
+        var receivedPhone = new PhoneController(new CustomRestClient()).GetPhone<Phone>(createdPhone.id).Phone;
+        phoneIdToDelete = receivedPhone.id;
+        var deletedPhone = new PhoneController(new CustomRestClient()).DeletePhone<RestResponse>(phoneIdToDelete);
```
Comment (dnnyhlwdplc) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/PhoneCreationTests.cs:30]: removing phone could be added in tear down. Here if removal fail you won't be able to verify the purpose of the test - CheckIfPossibleToAddNewPhoneObjectTest. Even if request will pass and return correct info.

File: EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/PhoneCreationTests.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        var isPhoneDeleted = new PhoneController(new CustomRestClient()).GetPhone<RestResponse>(phoneIdToDelete);
+        Assert.Multiple(() =>
```
Comment (dnnyhlwdplc) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.API.Tests/PhoneCreationTests.cs:32]: Nice usage of assert multiple!

**Review by dnnyhlwdplc — APPROVED** [review]

File: EpamWebSiteTestingFramework/Epam.TestAutomation.API/BaseController.cs (hunk 2 above)
Comment (katherinaGo) [reply EpamWebSiteTestingFramework/Epam.TestAutomation.API/BaseController.cs]: put!!! fixed

**Pull request merged**